Mathlib/CategoryTheory/Sites/Coverage.lean

```diff
@@ -0,0 +1,364 @@
+/-
+Copyright (c) 2023 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz
+-/
+
+import Mathlib.CategoryTheory.Sites.SheafOfTypes
+
+/-!
+
+# Coverages
+
+A coverage `K` on a category `C` is a set of presieve associated to every object `X : C`,
+called "covering presieves".
+This collection must satisfy a certain "pullback compatibility" condition, saying that
+whenever `S` is a covering presieve on `X` and `f : Y ⟶ X` is a morphism, then there exists
+some covering sieve `T` on `Y` such that `T` factors through `S` along `f`.
+
+The main difference between a coverage and a Grothendieck pretopology is that we *do not*
+require `C` to have pullbacks.
+This is useful, for example, when we want to consider the Grothendieck topology on the category
+of extremally disconnected sets in the context of condensed mathematics.
```

Comment:
What do you think of also mentioning the example of "(sub)basis for a topology"?

Reply:
Good idea! I'll add this later.